engine/src/main/java/com/google/android/fhir/sync/upload/Uploader.kt

```diff
@@ -38,67 +43,131 @@ import timber.log.Timber
  * 4. processing the responses from the server and consolidate any changes (i.e. updates resource
  *    IDs).
  */
-internal class Uploader(private val dataSource: DataSource) {
-  private val patchGenerator = PerResourcePatchGenerator
-  private val requestGenerator = TransactionBundleGenerator.getDefault()
-
+internal class Uploader(
+  private val dataSource: DataSource,
+  private val patchGenerator: PatchGenerator,
+  private val requestGenerator: UploadRequestGenerator,
+) {
   suspend fun upload(localChanges: List<LocalChange>): UploadSyncResult {
     val patches = patchGenerator.generate(localChanges)
-    val requests = requestGenerator.generateUploadRequests(patches)
+    val mappedRequests = requestGenerator.generateUploadRequests(patches)
```

Comment:
What do you think about doing the following to simplify:

* Revert `patchGenerator.generate(localChanges)` to return a `List<Patch>`
*  Revert `generateUploadRequests` back to `generateUploadRequests(patches: List<Patch>): List<UploadRequest>`
*  Let's do the chunking of the transaction bundles here in the uploader
```suggestion
 val mappedRequests = when (requestGenerator) {
      is UrlRequestGenerator -> requestGenerator.generateUploadRequests(patches)
      is TransactionBundleGenerator -> {
        patches.chunked(requestGenerator.generatedBundleSize)
          .flatMap { requestGenerator.generateUploadRequests(it) }
      }
      else -> error("")
    }
```


These changes I feel will allow us to simplify and get rid of the `PatchGeneratorOutput` class. wdyt?

Reply:
I think these changes are fine as they hep us keep our uploader class clean.